dango/testing/tests/lending.rs

```diff
@@ -0,0 +1,473 @@
+#[test_case(true; "with recipient arg")]
+fn withdraw_works(use_recipient: bool) {
+    let (mut suite, mut accounts, _codes, contracts) = setup_test();
+
+    let recipient = if use_recipient {
+        Some(accounts.owner.address())
+    } else {
+        None
+    };
+
+    // First deposit
+    suite
+        .execute(
+            &mut accounts.relayer,
+            contracts.lending,
+            &lending::ExecuteMsg::Deposit { recipient: None },
+            Coins::one(USDC.clone(), 123).unwrap(),
+        )
+        .should_succeed();
+    let lp_denom =
+        Denom::from_parts([NAMESPACE.to_string(), "lp".to_string(), USDC.to_string()]).unwrap();
+    suite
+        .query_balance(&accounts.relayer.address(), lp_denom.clone())
+        .should_succeed_and_equal(Uint128::new(123));
+
+    let balance_before = suite
+        .query_balance(
+            &recipient.unwrap_or(accounts.relayer.address()),
+            USDC.clone(),
+        )
+        .unwrap();
+
+    // Now withdraw
+    suite
+        .execute(
+            &mut accounts.relayer,
+            contracts.lending,
+            &lending::ExecuteMsg::Withdraw { recipient },
+            Coins::one(lp_denom.clone(), 123).unwrap(),
+        )
+        .should_succeed();
+
+    // Ensure LP token was burned from withdrawer.
+    suite
+        .query_balance(&accounts.relayer.address(), lp_denom)
+        .should_succeed_and_equal(Uint128::new(0));
+
+    // Ensure balance was added to recipient.
+    suite
+        .query_balance(
+            &recipient.unwrap_or(accounts.relayer.address()),
+            USDC.clone(),
+        )
+        .should_succeed_and_equal(balance_before + Uint128::new(123));
+}
+
+#[test]
+fn non_margin_accounts_cant_borrow() {
+    let (mut suite, mut accounts, _codes, contracts) = setup_test();
+
+    suite
+        .execute(
+            &mut accounts.relayer,
+            contracts.lending,
+            &lending::ExecuteMsg::Borrow {
+                coins: Coins::new(),
+            },
+            Coins::new(),
+        )
+        .should_fail_with_error("Only margin accounts can borrow");
+}
+
+#[test]
+fn borrowing_works() {
+    let (mut suite, mut accounts, codes, contracts) = setup_test();
+
+    // Create a margin account.
+    let mut margin_account = accounts
+        .relayer
+        .register_new_account(
+            &mut suite,
+            contracts.account_factory,
+            codes.account_margin.to_bytes().hash256(),
+            AccountParams::Margin(single::Params {
+                owner: accounts.relayer.username.clone(),
+            }),
+            Coins::new(),
+        )
+        .unwrap();
+
+    // Try to borrow from the margin account, should succeed fail as no coins are deposited
+    suite
+        .execute(
+            &mut margin_account,
+            contracts.lending,
+            &lending::ExecuteMsg::Borrow {
+                coins: Coins::from(Coin::new(USDC.clone(), 100).unwrap()),
```

Comment:
Use can you `Coins::one` to create a `Coins` consisting of only one coin.

Reply:
Ahh nice, thanks!